gms/gossiper.cc

```diff
@@ -1285,6 +1285,20 @@ future<> gossiper::do_gossip_to_unreachable_member(gossip_digest_syn message) {
                     addrs.insert(x.first);
                 }
             }
+            // The unreachable endpoints may be reachable via seed
```

Comment:
I meant to talk to nodes specified in seeds with some probability like we do in  do_gossip_to_unreachable_member. But no to mess up do_gossip_to_unreachable_member. 

Do it in a separate place.  Something like:
```
// Talk to nodes specified in seeds list with some probability in case nodes have changed ip address during restart 
// The node can only learn the new ip addresses from the seed list. 
do_gossip_to_intital_contact_points() {
       if (get_random() < 0.1) {
            for node in seeds list {
                   if (node is not alive) {
                        talk to node
                   }
            }
        }
}
```

Reply:
What is the difference with what is already done?
fFirst of all, we don't need to talk to seeds if the cluster is full and healthy. We need to do it only if the cluster has some dead nodes. Second, there is no point in maintaining two lists - the seed node can already be contacted with as a dead node by the do_gossip_to_unreachable_nodes(), so by having a separate function we will be doing double work. Finally, the choice of probability is arbitrary.

Reply:
The whole point of talking to node in  seeds list is that it might contain ip the node should talk to but it does not know it before.

I do not want to make it depends on _unreachable_endpoints logic and we can use different probability than the one used in do_gossip_to_unreachable_member to send to seed nodes.  That's why I suggested to do it outside do_gossip_to_unreachable_member.

In normal cases, even if we do it outside do_gossip_to_intital_contact_points, it is a noop, since all nodes are up.

In case some node are down, it does not matter we attempt to talk the same node in both do_gossip_to_unreachable_member and do_gossip_to_intital_contact_points. They use different probability anyway.

Reply:
I responded to this comment at the end of the pull request, please check it out.